packages/console/src/ds-components/TextInput/MultiInput.tsx

```diff
@@ -0,0 +1,171 @@
+};
+
+type Props = {
+  className?: string;
+  values: Option[];
+  onChange: (values: Option[]) => void;
+  error?: string | boolean;
+  placeholder?: AdminConsoleKey;
+  renderOption?: (option: Option) => React.ReactNode;
+};
+
+// RegExp to domain string.
+// eslint-disable-next-line prefer-regex-literals
+const domainRegExp = new RegExp('\\S+[\\.]{1}\\S+');
+
+function MultiInput({
+  className,
+  values,
+  onChange,
+  error,
+  placeholder,
+  renderOption = ({ value }) => value,
+}: Props) {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [isInputFocused, setIsInputFocused] = useState(false);
+  const [focusedValueId, setFocusedValueId] = useState<Nullable<string>>(null);
+  const [currentValue, setCurrentValue] = useState('');
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+
+  const isCurrentValueValidDomain = useMemo(() => {
+    return domainRegExp.test(currentValue);
+  }, [currentValue]);
+
+  const handleAdd = (value: string) => {
+    const newValues = [...values, { value, id: generateStandardShortId() }];
+    onChange(newValues);
+    setCurrentValue('');
+    inputRef.current?.focus();
+  };
+
+  const handleDelete = (option: Option) => {
+    onChange(values.filter(({ id }) => id !== option.id));
+  };
+
+  // https://exogen.github.io/blog/focus-state/
+  useEffect(() => {
+    if (document.hasFocus() && inputRef.current?.contains(document.activeElement)) {
+      setIsInputFocused(true);
+    }
+  }, []);
+
+  return (
+    <div
+      className={classNames(
+        styles.input,
+        styles.multiple,
+        isInputFocused && styles.inputFocused,
+        Boolean(error) && styles.error,
+        className
+      )}
+      role="button"
+      tabIndex={0}
+      onKeyDown={onKeyDownHandler(() => {
+        inputRef.current?.focus();
+      })}
+      onClick={() => {
+        inputRef.current?.focus();
+      }}
+    >
+      {values.map((option) => {
+        return (
+          <Tag
+            key={option.id}
+            variant="cell"
+            className={classNames(
+              styles.tag,
+              option.status && styles[option.status],
+              option.id === focusedValueId && styles.focused
+            )}
+            onClick={(event) => {
+              event.stopPropagation();
+            }}
+          >
+            {renderOption(option)}
+            <IconButton
+              className={styles.delete}
+              size="small"
+              onClick={() => {
+                handleDelete(option);
+              }}
+              onKeyDown={onKeyDownHandler(() => {
+                handleDelete(option);
+              })}
+            >
+              <Close className={styles.close} />
+            </IconButton>
+          </Tag>
+        );
+      })}
+      <input
+        ref={inputRef}
+        type="text"
+        placeholder={conditional(values.length === 0 && placeholder && String(t(placeholder)))}
```

Comment:
The condition should not be needed, as placeholder only works when the value is empty

Reply:
Should not show place holder if there are already `values`